_server/tests/component/BusinessLogicTest.cpp

```diff
@@ -71,6 +72,33 @@ TEST_F(BusinessLogicTest, UsersModule_LoginUser)
     ASSERT_TRUE(callbackInvoked);
 }
 
+TEST_F(BusinessLogicTest, UsersModule_addRole)
+{
+    RequestData requestData{
+        .module = "users",
+        .submodule = "roles",
+        .method = "GET",
+        .resourceId = "",
+        .dataset = {}};
+
+    std::vector<Role> expectedUsers{};
+    EXPECT_CALL(*rolesRepositoryMock, getAll())
+        .WillOnce(Return(expectedUsers));
+    // Prepare the callback and its expectation
+    bool callbackInvoked = false;
+    auto callback = [&callbackInvoked](const ResponseData &responseData)
+    {
+        callbackInvoked = true;
+        ASSERT_TRUE(responseData.dataset.find(Keys::_ERROR) == responseData.dataset.end());
+    };
+
+    // Execute the task
+    businessLogic->executeTask(requestData, callback);
+
+    // Verify the callback was invoked
+    ASSERT_TRUE(callbackInvoked);
```

Comment:
what it the purpose of this check? callbackInvoked will be always true at this part of code, won't it?

Reply:
done